Sources/Testing/ExitTests/WaitFor.swift

```diff
@@ -80,7 +80,7 @@ func wait(for pid: consuming pid_t) async throws -> ExitCondition {
 }
 #elseif SWT_TARGET_OS_APPLE || os(Linux) || os(FreeBSD) || os(OpenBSD)
 /// A mapping of awaited child PIDs to their corresponding Swift continuations.
-private let _childProcessContinuations = Locked<[pid_t: CheckedContinuation<ExitCondition, any Error>]>()
+private let _childProcessContinuations = LockedWith<pthread_mutex_t, [pid_t: CheckedContinuation<ExitCondition, any Error>]>()
```

Comment:
Why do you need to specifically use LockedWith and specify pthread_mutex here instead of Locked?

Reply:
Because we explicitly need a `pthread_mutex_t` so we can use a `pthread_cond_t` with it. In the future, we might want to e.g. change the Linux default lock to a futex, but this can't change.